src/main/java/org/folio/rest/impl/TenantRefAPI.java

```diff
@@ -41,37 +40,56 @@ Future<Integer> loadData(TenantAttributes attributes, String tenantId,
       .compose(r -> new KafkaAdminClientService(vertxContext.owner())
         .createKafkaTopics(CirculationStorageKafkaTopic.values(), tenantId))
       .compose(r -> super.loadData(attributes, tenantId, headers, vertxContext))
-      .compose(superRecordsLoaded -> {
-        log.info("Initializing of tenant's data");
-        Vertx vertx = vertxContext.owner();
-        Promise<Integer> promise = Promise.promise();
-        TenantLoading tl = new TenantLoading();
-        tl.withKey(REFERENCE_KEY).withLead(REFERENCE_LEAD)
-          .withIdContent()
-          .add("loan-policy-storage/loan-policies")
-          .add("request-policy-storage/request-policies")
-          .add("patron-notice-policy-storage/patron-notice-policies")
-          .add("staff-slips-storage/staff-slips")
-          .withIdRaw()
-          .add("circulation-rules-storage")
-          .withIdContent()
-          .add("cancellation-reason-storage/cancellation-reasons")
-          .withKey(SAMPLE_KEY).withLead(SAMPLE_LEAD)
-          .add("loans", "loan-storage/loans")
-          .add("requests", "request-storage/requests")
-          .add("circulation-settings-storage/circulation-settings");
+      .compose(r -> loadData(attributes, headers, vertxContext))
+      .compose(r -> registerModule(headers, vertxContext))
+      .mapEmpty();
+  }
+
+  private Future<Integer> loadData(TenantAttributes attributes, Map<String, String> headers,
+      Context vertxContext) {
 
+    log.info("Initializing of tenant's data");
+    TenantLoading tl = new TenantLoading();
+    tl.withKey(REFERENCE_KEY).withLead(REFERENCE_LEAD);
+    if (isNew(attributes, "14.0.0")) {
```

Comment:
What should we do when we want to change reference data in newer versions? When we update this line ("14.0.0" -> new version), the reference data is going to be updated anyway. However, we'll be in control of that and can let libraries know through Release notes. Was that the intention?